tests/test_entities.py

```diff
@@ -2899,33 +2905,64 @@ def test_read(self):
         self.read_json_pacther.stop()
         self.read_pacther.stop()
 
-    def test_delete_puppetclass(self):
+    def test_add_func_with_id(self):
         """Check that helper method is sane.
 
         Assert that:
 
         * Method has a correct signature.
         * Method calls `client.*` once.
         * Method passes the right arguments to `client.*` and special
-            argument 'puppetclass_id' removed from data dict.
+            argument 'ansible_role_id' removed from data dict.
         * Method calls `entities._handle_response` once.
         * The result of `_handle_response(…)` is the return value.
+        """
+        entity = self.entity
+        entity.id = 1
+        func_param_dict = {entity.add_ansible_role: 'ansible_role_id'}
+        for func in list(func_param_dict.keys()):
```

Comment:
No need to cast to list, keys() returns the list object internally, you even can use the items() method and catch the key and value.

Reply:
updated